### ISSUE TITLE
this is not replaced by __$ctx in custom predicates in production mode

### ISSUE THREAD

**Issue by arikon** [bem/bem-xjst#64]
Template (see the last template):

```js
block('tumbler')(
    tag()('span'),
    js()(function() {
        return this.extend({
            live: false
        }, this.ctx.js)
    }),
    def()(
        match(function() {
            return true
        })(function() {
            var id = this.generateId(),
                leftId = 'left' + id,
                rightId = 'right' + id,
                sideId = null,
                content = this.ctx.content;
            if (content && content.length > 0) {
                sideId = this.ctx.content[0].side === 'left' ? leftId : rightId
            };
            applyNext({
                _name: (this.ctx.name || ((this.ctx.name = id))),
                _leftId: leftId,
                _rightId: rightId,
                _sideId: sideId
            })
        }),
        match(function() {
            return !((this.ctx.mods || {})).theme
        })(function() {
            applyNext({
                'ctx.mods': this._.extend((this.ctx.mods || {}), {
                    theme: 'normal'
                })
            })
        }),
        match(function() {
            return ((this.ctx.mods || {})).checked === 'yes'
        })(function() {
            applyNext({
                _checked: 'checked'
            })
        }),
        match(function() {
            return ((this.ctx.mods || {})).disabled === 'yes'
        })(function() {
            applyNext({
                _disabled: 'disabled'
            })
        })
    ),
    content().match(function() {
        return (this._.isArray(this.ctx.content) && this.ctx.content.length === 1)
    })(function() {
        var content = this.ctx.content;
        if (content[0].side !== 'left') {
            return [{
                elem: 'option',
                side: 'left'
            }, content]
        };
        return [content, {
            elem: 'option',
            side: 'right'
        }]
    }),
    content().match(function() {
        return this.ctx.content === undefined
    })(function() {
        return [{
            elem: 'option',
            side: 'left'
        }, {
            elem: 'option',
            side: 'right'
        }]
    })
)
```

Compiled to (a part of):

```js
            var __$t = !$$elem;
            if (__$t) {
                if (undefined === this.ctx.content) {
                    return [ {
                        elem: "option",
                        side: "left"
                    }, {
                        elem: "option",
                        side: "right"
                    } ];
                }
                if (__$ctx._.isArray(__$ctx.ctx.content) && __$ctx.ctx.content.length === 1) {
                    var __$r = __$b37(__$ctx, __$ref);
                    if (__$r !== __$ref) return __$r;
                }
            }
```

The problem is in the `undefined === this.ctx.content` that should be `undefined === __$ctx.ctx.content`.

The error during templates apply is: `TypeError: Cannot read property 'content' of undefined`

**Comment by vkz** [comment]
found a fix, will pr shortly

**Comment by arikon** [comment]
@indutny This fix works https://github.com/veged/xjst/issues/45

**Comment by indutny** [comment]
Great, released in xjst@1.5.2
